langroid/language_models/openai_gpt.py

```diff
@@ -730,37 +820,41 @@ def _process_stream_event(
             if not silent:
                 sys.stdout.write(Colors().GREEN + event_text)
                 sys.stdout.flush()
-                self.config.streamer(event_text)
+            await self.config.streamer(event_text)
```

Comment:
This should be inside the `if not silent:` block, and similarly elsewhere in this code.

Reply:
Also, calling `self.config.streamer(event_text)` results in an error when we directly use the LLM (i.e. not through a `ChatAgent`), since the default setting of `self.config.streamer` is `noop_fn()` which is a sync fn.

Reply:
One solution could be to have a separate `config.streamer_async` in `LLMConfig` defined as an async version of noop_fn, and use this streamer inside `_process_event_stream_async`